### PR TITLE
Fix Azure.Monitor.Query live tests

### PR DESCRIPTION
**Pull request by pakrym**
Fixed a bunch of issues around seeding the test data, service bugs and unreability.

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ public void CanSetServiceTimeout_Mocked()
                 Assert.True(message.Request.Headers.TryGetValue("prefer", out preferHeader));
                 networkOverride = message.NetworkTimeout;
 
-                return new MockResponse(500);
+                return new MockResponse(403);
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs:23]: We would retry 500s making the test take longer.

File: sdk/monitor/Azure.Monitor.Query/tests/MetricsTestData.cs [hunk 2]
```diff
@@ -44,17 +44,25 @@ public async Task InitializeAsync()
             _initialized = true;
             var metricClient = new MetricsQueryClient(_testEnvironment.MetricsEndpoint, _testEnvironment.Credential);
 
+            await SendData();
```
Comment (pakrym) [inline sdk/monitor/Azure.Monitor.Query/tests/MetricsTestData.cs:47]: Send data only once .

**Review by christothes — APPROVED** [review]

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs [hunk 3]
```diff
@@ -34,7 +34,7 @@ public void CanSetServiceTimeout_Mocked()
             }));
 
             Assert.AreEqual("wait=600", preferHeader);
-            Assert.AreEqual(TimeSpan.FromMinutes(10), networkOverride);
+            Assert.AreEqual(TimeSpan.FromMinutes(10).Add(TimeSpan.FromSeconds(15)), networkOverride);
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs:37]: Actually I'm a fan of hardcoding values into tests. Makes it more obvious what's being affected by changes.

[the remark this answers is not in the document]
Comment (christothes) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs:37]: Fair - the other perspective is that if this ever changes, how would we know which value is wrong? Is there a reason this test felt it was important to be 15, or should it just validate that it is consistent with whatever the client sets it to?

File: sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs [hunk 4]
```diff
@@ -226,7 +227,8 @@ private async Task<Response<LogsQueryResult>> ExecuteAsync(string workspaceId, s
 
             if (options?.ServerTimeout != null)
             {
-                message.NetworkTimeout = options.ServerTimeout;
+                // Offset the service timeout a bit to make sure we have time to receive the response.
+                message.NetworkTimeout = options.ServerTimeout.Value.Add(_networkTimeoutOffset);
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs:231]: I think we always want to add it, if the service generates the response in requested time we shouldn't abort the receiving. 15 sec seemed like a safe value considering the max timeout it 10 minutes.

File: sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/tests/LogsQueryClientClientTests.cs:37]: Fair. I'll add a comment

File: sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQueryResultInternal.cs [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+                try
+                {
+                    using var document = JsonDocument.Parse(element.GetString());
+                    element = document.RootElement.Clone();
+                }
+                catch
+                {
+                    // ignore
+                }
+            }
```
[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/monitor/Azure.Monitor.Query/src/Models/LogsBatchQueryResultInternal.cs:35]: There is now.

**Pull request merged**